runtime/api/src/main/java/org/qbicc/runtime/SerializedAs.java

```diff
@@ -0,0 +1,20 @@
+package org.qbicc.runtime;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * An annotation to indicate that when an instance or static field is
+ * serialized to the qbicc initial heap that whatever value was stored
+ * in the field in the build time heap should be replaced by the
+ * value provided by the annotation in the initial runtime heap.
+ * The replacement value will be that supplied by the value method
+ * that matches the type of the field.
+ */
+@Retention(RetentionPolicy.CLASS)
+@Target(ElementType.FIELD)
+public @interface SerializedAs {
+    long value();
+}
```

Comment:
Not trying to be difficult (honest!)... but this should either be `SerializedAsLong` with a property named `value` (which we may eventually supplement with additional annotations like `SerializedAsString` or whatever) or `SerializedAs` with a property named `longValue` (which we may eventually supplement with defaulted properties named `intValue`, `stringValue`, etc.`). I think the former option (additional annotations in the future) is probably a bit more laborious but it might give us a bit more user-friendly flexibility (for example we could eventually have an annotation that gives a factory class and method name which are called to produce the value).

Reply:
No worries... the current implementation is probably overly cute and I didn't write the comment to explain it.  Worth the time iterating on this, because it is going to be part of the user experience. 

The current single annotation takes the 64 bit long `value` and based on the type of the annotated field interprets those bits as a constant of that type.  I think this is pretty sensible for boolean, byte, char, short, int, and long.  It has a well-defined meaning for float and double, but is of questionable usability (as you can't actually do `Double.toLongBits(1.0)` for example in an annotation value).  It doesn't support String (or Class).

If we don't mind having 9 separate annotations, it wouldn't be that much work to just define them for each primitive type + String (`SerializeAsInt`, `SerializeAsChar`, etc).  I think that is probably more usable than a single annotation with 9 properties.

We could also do an intermediate design where we had four SerializeAs annotations: Boolean, FloatingPoint, Integral, and String.

Reply:
Revised to separate Boolean, Integral, and FloatingPoint types.  Held off on String because it seemed like that would be better to handle under a general replacement mechanism for Objects (which I don't know how to do yet or if we will really need it) as opposed to a special case.